st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -579,13 +579,42 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        if cfg.CONF.ssh_runner.use_ssh_config:
+            conninfo_host_ssh_config = self._get_conninfo_from_ssh_config_for_host(host)
+            conninfo.update(conninfo_host_ssh_config)
```

Comment:
Actually, now that I think of it - is the current behavior the desired one?

It looks like currently values  from the config have precedence over other values (provided as action parameters)? It seems like values which are provided as parameters should have precedence over the ones provided in the config.

In any case, it's something we need to document and have good tests for so there will be no surprises.